chevah/empirical
# [#3510] Allow nested HTTP server contexts.

Scope
=====

This allows running nested HTTPServer context.


Changes
=======

The context was updated to restore the state found on entering the context.

As a drive-by change I have updated the test code to have 100% coverage.... codecov is still broken and will not fully report an unified diff from all builders.


How to try and test the changes
===============================

reviewers: @brunogola

Check that changes make sense
